mlir-sycl/lib/Dialect/IR/SYCLOpsTypes.cpp

```diff
@@ -215,3 +216,11 @@ mlir::sycl::VecType::verify(llvm::function_ref<InFlightDiagnostic()> EmitError,
 
   return success();
 }
+
+unsigned mlir::sycl::getDimensions(mlir::Type Type) {
+  if (auto MemRefTy = Type.dyn_cast<mlir::MemRefType>())
+    Type = MemRefTy.getElementType();
+  return TypeSwitch<mlir::Type, unsigned>(Type)
```

Comment:
Should add a `Default` to the type switch. The Default could issue an assertion and return 0. Or could issue an llvm_unreachable... your choice.

Reply:
No adding a default when a return vale is expected = implicit `llvm_unreachable` :)